lang/csharp/src/apache/main/CodeGen/CodeGen.cs

```diff
@@ -766,6 +767,13 @@ internal static string getType(Schema schema, bool nullible, ref bool nullibleEn
                         return CodeGenUtil.Object;
                     else
                         return getType(nullibleType, true, ref nullibleEnum);
+
+                case Schema.Type.Logical:
+                    var logicalSchema = schema as LogicalSchema;
+                    if (null == logicalSchema)
+                        throw new CodeGenException("Unable to cast schema into a logical schema");
+                    return logicalSchema.LogicalType.GetCSharpTypeName(nullible);
```

Comment:
Could we allow users to opt-out of the logical type support in the code generation? Would that break a lot of things?

Reply:
The generated class would be semantically incorrect though? Your schema would define a property but the generated class wouldn't. That would break all sorts of things?

Even if you dropped to just using the underlying types? Not sure how useful that is though - if my intention is not to have a logical type, then surely you would need to define the schema the same way.